docs/components/geolocate-control.md

```diff
@@ -0,0 +1,42 @@
+# Geolocate Control
+
+This is a React equivalent of Mapbox's [GeolocateControl](https://www.mapbox.com/mapbox-gl-js/api/#geolocatecontrol).
+
+```js
+import React, { Component } from "react";
+import ReactMapGL, {GeolocateControl} from "react-map-gl";
+
+class Map extends Component {
+  render() {
+    const { viewport, updateViewport } = this.props;
+    return (
+      <ReactMapGL {...viewport} onViewportChange={updateViewport}>
+        <div style={{ position: "absolute", right: 0 }}>
+          <GeolocateControl 
+            positionOptions={{enableHighAccuracy: true}}
+            trackUserLocation={true}
+            onViewportChange={updateViewport}
+          />
+        </div>
+      </ReactMapGL>
+    );
+  }
+}
+```
+
+## Properties
+
+Accepts all the options of [Mapbox GeolocatControl](https://docs.mapbox.com/mapbox-gl-js/api/#geolocatecontrol)
```

Comment:
List these props for clarity.
Mapbox may add new options to this control in future releases and we still need to update the component to support them.